src/firewall/iptables.rs

```diff
@@ -233,11 +255,99 @@ impl firewall::FirewallDriver for IptablesDriver {
 
     fn teardown_port_forward(
         &self,
-        _container_id: &str,
-        _port_mappings: Vec<types::PortMapping>,
-        _container_ip: &str,
+        network: Network,
+        container_id: &str,
+        port_mappings: Vec<types::PortMapping>,
+        network_name: &str,
+        id_network_hash: &str,
+        options: &PerNetworkOptions,
     ) -> Result<(), Box<dyn Error>> {
-        todo!();
+        let container_ips = options.static_ips.as_ref().ok_or_else(|| {
+            std::io::Error::new(std::io::ErrorKind::Other, "no container ip provided")
+        })?;
+        let networks = &network.subnets.as_ref().ok_or_else(|| {
+            std::io::Error::new(std::io::ErrorKind::Other, "no network address provided")
+        })?;
+        let container_network_address = networks[0].subnet;
+        let network_dn_chain_name = CONTAINER_DN_CHAIN.to_owned() + id_network_hash;
+        let comment_dn_network_cid = format!(
+            "-m comment --comment 'dnat name: {} id: {}'",
+            network_name, container_id
+        );
+        let network_chain_name = CONTAINER_CHAIN.to_owned() + id_network_hash;
+        let container_ip = container_ips[0];
+        // First delete any container specific rules
+        // POSTROUTING
+        let comment_network_cid = format!(
+            "-m comment --comment 'name: {} id: {}'",
+            network_name, container_id
+        );
+        remove_if_exists(
```

Comment:
What chains are you actually removing here? I think you missed the network's NAT chain

Reply:
that code is in a different PR as stated in the commit message